src/Map/ProspectorOverlayLayer.cs

```diff
@@ -77,294 +135,172 @@ public ProspectorOverlayLayer(ICoreAPI api, IWorldMapManager mapSink) : base(api
-
-                        RebuildMap(true);
-                    }
-                    catch (FormatException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    catch (ArgumentException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    break;
-                case "sethighheatcolor":
-                    try
-                    {
-                        var newColor = TrySetRGBAValues(args, _config.BorderColor);
-                        _config.HighHeatColor = newColor;
-                        _config.Save(api);
+                foreach (ProspectInfo info in information)
+                {
+                    _prospectInfos[info.ChunkCoordinate] = info;
+                    var newComponent = new ProspectorOverlayMapComponent(_clientApi, info.ChunkCoordinate, info.GetMessage(), _colorTextures[(int)GetRelativeDensity(info)]);
+                    _components[info.ChunkCoordinate] = newComponent;
+                }
+                _prospectInfos.HasChanged = true;
+            }
+        }
+        #endregion
 
-                        RebuildMap(true);
-                    }
-                    catch (FormatException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    catch (ArgumentException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    break;
-                case "setbordercolor":
-                    try
-                    {
-                        var newColor = TrySetRGBAValues(args, _config.BorderColor);
-                        _config.BorderColor = newColor;
-                        _config.Save(api);
+        #region Commands/Events
 
-                        RebuildMap(true);
-                    }
-                    catch (FormatException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    catch (ArgumentException e)
-                    {
-                        _clientApi.ShowChatMessage(e.Message);
-                    }
-                    break;
-                case "setborderthickness":
-                    var newThickness = args.PopInt(2).Value;
-                    _config.BorderThickness = newThickness;
-                    _config.Save(api);
+        private TextCommandResult OnShowOverlayCommand(TextCommandCallingArgs args) 
+        {
+            // Parser count is 0 when only calling .pi command.
+            if (args.Parsers.Count == 0 || args.Parsers[0].IsMissing)
+                _config.RenderTexturesOnMap = !_config.RenderTexturesOnMap;
+            else
+                _config.RenderTexturesOnMap = (bool) args.Parsers[0].GetValue();
+            _config.Save(api);
```

Comment:
i think a rebuildMap is missing here. 
```suggestion
            _config.Save(api);
            RebuildMap();
```
A weird side-effect when its missing is the following at least for me 
![image](https://user-images.githubusercontent.com/5238284/234052702-7a086207-17f9-4d2d-9c72-5ff66668bf2c.png)



```
24.4.2023 18:02:33 [Client Error] Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.JsonTesselator.AddJsonModelDataToMesh(MeshData sourceMesh, Int32 lodlevel, TCTCache vars, IMeshPoolSupplier poolSupplier, Int32 atlasTextureId, Single[] tfMatrix) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator\JsonTesselator.cs:line 484
   at Vintagestory.Client.NoObf.JsonTesselator.doMesh(TCTCache vars, MeshData sourceMesh, Int32 atlasTextureId, Int32 lodLevel) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator\JsonTesselator.cs:line 212
   at Vintagestory.Client.NoObf.JsonTesselator.Tesselate(TCTCache vars) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator\JsonTesselator.cs:line 161
   at Vintagestory.Client.NoObf.ChunkTesselator.TesselateBlock(Block block, Int32 lX, Int32 faceflags, Int32 posX, Int32 posZ, Int32 drawType) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator.cs:line 960
   at Vintagestory.Client.NoObf.ChunkTesselator.TesselateBlock(Int32 index3d, Int32 extIndex3dBase, Int32 lX, Int32 baseX, Int32 posZ) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator.cs:line 894
   at Vintagestory.Client.NoObf.ChunkTesselator.BuildBlockPolygons(Int32 chunkX, Int32 chunkY, Int32 chunkZ) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Render\Tesselation\ChunkTesselator.cs:line 734
```

Reply:
I can't reproduce this but I added the RebuildMap